### PR TITLE
Fixes #1038 Failed to create buckets due to incorrect RAM calculation

### LINKED ISSUE #1038: Failed to create buckets due to incorrect RAM calculation
In [cen7-sync-gateway-functional-tests-base-cc-specific-commit/8](http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-base-cc-specific-commit/8/consoleFull)

```
16:43:43 GET http://ec2-54-86-109-92.compute-1.amazonaws.com:8091/pools/default/buckets 200
16:43:43 GET http://ec2-54-86-109-92.compute-1.amazonaws.com:8091/pools/default/buckets 200
16:43:43 Creating buckets: [u'data-bucket']
16:43:43 total_avail_ram_mb: 7062
16:43:43 effective_avail_ram_mb: 5137
16:43:43 Creating bucket data-bucket with RAM 5137
16:43:43 POST http://ec2-54-86-109-92.compute-1.amazonaws.com:8091/pools/default/buckets 400
16:43:43 ---------------------------- Captured stderr setup -----------------------------
16:43:43  [WARNING]: Consider using file module with state=absent rather than running rm
16:43:43  [WARNING]: While constructing a mapping from /var/jenkins/workspace/cen7-sync-
16:43:43 gateway-functional-tests-base-cc-specific-
16:43:43 commit/libraries/provision/ansible/playbooks/install-couchbase-server-
16:43:43 package.yml, line 49, column 7, found a duplicate dict key (shell). Using last
16:43:43 defined value only.
16:43:43  ERROR at setup of test_attachment_revpos_when_ancestor_unavailable_active_revision_doesnt_share_ancestor[sync_gateway_default] 
16:43:43 
```

This cluster is using Couchbase Server nodes with different sizes, and due to this code:

```
    def _get_total_ram_mb(self):
        """
        Call the Couchbase REST API to get the total memory available on the machine. RAM returned is in mb
        """
        resp = self._session.get("{}/pools/default".format(self.url))
        resp.raise_for_status()
        resp_json = resp.json()

        # Workaround for https://github.com/couchbaselabs/mobile-testkit/issues/709
        # where some node report mem_total = 0. Loop over all the nodes and find highest val
        mem_total_highest = 0
        for node in resp_json["nodes"]:
            mem_total = node["systemStats"]["mem_total"]
            if mem_total > mem_total_highest:
                mem_total_highest = mem_total

        total_avail_ram_mb = int(mem_total_highest / (1024 * 1024))
        log_info("total_avail_ram_mb: {}".format(total_avail_ram_mb))
        return total_avail_ram_mb
```

It's choosing the node with the **largest** amount of RAM and then deciding how to size the bucket based on that.  However, it ends up trying to use too much RAM and couchbase server refuses to create the bucket.

```
Creating bucket data-bucket with RAM 5137
```

When I tried creating a bucket with 5137MB of RAM in the Couchbase Web UI, it also failed.

## Proposed fix

`_get_total_ram_mb` should use the smallest non-zero `node["systemStats"]["mem_total"]` value from all of the nodes.

### PR DESCRIPTION
**Pull request by tleyden**
#### Fixes #1038.

- [x] Ran `flake8`

#### Changes proposed in this pull request:

- Fixes #1038

### CODE REVIEW

**Comment by tleyden** [comment]
Functional test run [cen7-sync-gateway-functional-tests-base-cc-specific-commit/9](http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-base-cc-specific-commit/9/) in progress

**Comment by tleyden** [comment]
> Functional test run cen7-sync-gateway-functional-tests-base-cc-specific-commit/9 in progress

This one caught a (bad) issue, but:

[cen7-sync-gateway-functional-tests-base-cc-specific-commit/10](http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-base-cc-specific-commit/10/console) seems to be doing better so far.

**Comment by tleyden** [comment]
Ok the fix definitely appears to have worked.

In my case I have three nodes:

1. 3.51 GB
1. 6.89 GB
1. 3.51 GB

The bucket was created with 6.94 GB, which matches up with the cluster quota given the replication factor (1)

screen shot 2017-03-13 at 6 10 12 pm

File: keywords/couchbaseserver.py [hunk 1]
```diff
@@ -183,14 +183,25 @@ def _get_total_ram_mb(self):
         resp_json = resp.json()
 
         # Workaround for https://github.com/couchbaselabs/mobile-testkit/issues/709
-        # where some node report mem_total = 0. Loop over all the nodes and find highest val
-        mem_total_highest = 0
+        # Later updated for https://github.com/couchbaselabs/mobile-testkit/issues/1038
+        # where some node report mem_total = 0. Loop over all the nodes and find the smallest non-zero val
+        mem_total_lowest = None
```
Comment (sethrosetter) [inline keywords/couchbaseserver.py]: Could we pull this out into a unit-testable method and add a unit test?

**Comment by sethrosetter** [comment]
@raghusarangapani Please review when you get a chance.

**Comment by tleyden** [comment]
@raghusarangapani anything blocking this?

**Pull request merged**